addons/sms/wizard/sms_composer.py

```diff
@@ -21,7 +21,12 @@ def default_get(self, fields):
             return result
 
         result['res_model'] = result.get('res_model') or self.env.context.get('active_model')
-        result['composition_mode'] = result.get('composition_mode') or 'comment'
+        result['composition_mode'] = result.get('composition_mode')
+
+        # For the context action creation from the sms template, if multiples ids => mass composition
+        if self.env.context.get('template_context_action') and self.env.context.get('active_ids') and len(self.env.context.get('active_ids')) > 1:
```

Comment:
I don't like that kind of "if / else" code that introduces a corner case ... it makes code hard to understand. At least use a context key with a meaning, like "default_composition_mode: guess" that would be used as a magic key to determine whether to use comment or mass mode.